src/features/expressions/expression-functions.ts

```diff
@@ -14,613 +13,635 @@ import { BaseLayoutNode } from 'src/utils/layout/LayoutNode';
+  },
+  stringLength: {
+    args: args(required(ExprVal.String)),
     returns: ExprVal.Number,
-  }),
-  commaContains: defineFunc({
-    impl(commaSeparatedString, stringToMatch) {
-      if (commaSeparatedString === null || stringToMatch === null) {
-        return false;
-      }
-
-      // Split the comma separated string into an array and remove whitespace from each part
-      const parsedToArray = commaSeparatedString.split(',').map((part) => part.trim());
-      return parsedToArray.includes(stringToMatch);
-    },
-    args: [ExprVal.String, ExprVal.String] as const,
+  },
+  commaContains: {
+    args: args(required(ExprVal.String), required(ExprVal.String)),
     returns: ExprVal.Boolean,
-  }),
-  lowerCase: defineFunc({
-    impl(string) {
-      if (string === null) {
-        return null;
-      }
-      return string.toLowerCase();
-    },
-    args: [ExprVal.String] as const,
+  },
+  lowerCase: {
+    args: args(required(ExprVal.String)),
     returns: ExprVal.String,
-  }),
-  upperCase: defineFunc({
-    impl(string) {
-      if (string === null) {
-        return null;
-      }
-      return string.toUpperCase();
-    },
-    args: [ExprVal.String] as const,
+  },
+  upperCase: {
+    args: args(required(ExprVal.String)),
     returns: ExprVal.String,
-  }),
-  _experimentalSelectAndMap: defineFunc({
-    args: [ExprVal.String, ExprVal.String, ExprVal.String, ExprVal.String, ExprVal.Boolean] as const,
-    impl(path, propertyToSelect, prepend, append, appendToLastElement = true) {
-      if (path === null || propertyToSelect == null) {
-        throw new ExprRuntimeError(this.expr, this.path, `Cannot lookup dataModel null`);
-      }
+  },
+  _experimentalSelectAndMap: {
+    args: args(
+      required(ExprVal.String),
+      required(ExprVal.String),
+      optional(ExprVal.String),
+      optional(ExprVal.String),
+      optional(ExprVal.Boolean),
+    ),
+    returns: ExprVal.String,
+  },
+} satisfies { [key: string]: AnyFuncDef };
+
+type Defs = typeof ExprFunctionDefinitions;
+type Names = keyof Defs;
+type Implementation<Name extends Names> = (
+  this: EvaluateExpressionParams,
+  ...params: ArgsToActual<Defs[Name]['args']>
+) => ExprValToActual<Defs[Name]['returns']> | null;
+
+export const ExprFunctionImplementations: { [K in Names]: Implementation<K> } = {
+  argv(idx) {
+    if (!this.positionalArguments?.length) {
+      throw new ExprRuntimeError(this.expr, this.path, 'No positional arguments available');
+    }
+
+    if (idx === null || idx < 0 || idx >= this.positionalArguments.length) {
+      throw new ExprRuntimeError(this.expr, this.path, `Index ${idx} out of range`);
+    }
+
+    return this.positionalArguments[idx];
+  },
+  value(key) {
+    const config = this.valueArguments;
+    if (!config) {
+      throw new ExprRuntimeError(this.expr, this.path, 'No value arguments available');
+    }
+
+    const realKey = (key ?? config.defaultKey) as string | null;
+    if (!realKey) {
```

Comment:
Kan du heller sjekke typen her i stedet for type assertion? Skulle gjerne helst blitt kvitt `any` i `ExprValueArgs`, men det var litt mer jobb enn jeg var forberedt på nå 😅 
```suggestion
    const realKey = (key ?? config.defaultKey) as string | null;
    if (typeof realKey !== 'string') {
```